bootstrap/file/file.go

```diff
@@ -0,0 +1,77 @@
+package file
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/interfaces"
+)
+
+func Load(path string, timeout time.Duration, provider interfaces.SecretProvider) ([]byte, error) {
+	var fileBytes []byte
+	var err error
+
+	parsedUrl, err := url.Parse(path)
+	if err != nil {
+		return nil, fmt.Errorf("Could not parse file path: %v", err)
+	}
+
+	if parsedUrl.Scheme == "http" || parsedUrl.Scheme == "https" {
+		client := &http.Client{
+			Timeout: timeout,
+		}
+		req, err := http.NewRequest("GET", path, nil)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to create new request for remote file: %s: %v", parsedUrl.Redacted(), err)
+		}
+
+		// Get httpheader secret
+		params := parsedUrl.Query()
+		edgexSecretName := params.Get("edgexSecretName")
+		if edgexSecretName != "" {
+			secrets, err := provider.GetSecret(edgexSecretName)
+			if err != nil {
+				return nil, err
+			}
+
+			// Set request header
+			if len(secrets) > 0 && secrets["type"] == "httpheader" {
+				if secrets["headername"] != "" && secrets["headercontents"] != "" {
+					req.Header.Add(secrets["headername"], secrets["headercontents"])
+				} else {
+					return nil, fmt.Errorf("Secret headername and headercontents can not be empty")
+				}
```

Comment:
could simplify this but it's ok:
```suggestion
				if secrets["headername"] == "" || secrets["headercontents"] == "" {
				        return nil, fmt.Errorf("Secret headername and headercontents can not be empty")
				} 
					req.Header.Add(secrets["headername"], secrets["headercontents"])
```